errbot/backends/irc.py

```diff
@@ -651,9 +716,7 @@ def rooms(self):
         :returns:
             A list of :class:`~IRCMUCRoom` instances.
         """
-
-        channels = self.conn.channels.keys()
-        return [IRCRoom(channel) for channel in channels]
+        return self._rooms
```

Comment:
`self._rooms` is a dictionary, but the API of `rooms()` says this should return a list of instances so this should be `return self._rooms.values()` instead.